ddtrace/tracer/option.go

```diff
@@ -109,13 +110,23 @@ func defaults(c *config) {
 		WithEnv(v)(c)
 	}
 	if v := os.Getenv("DD_SERVICE"); v != "" {
-		c.serviceName = v
+		WithService(v)
 		globalconfig.SetServiceName(v)
 	} else {
-		c.serviceName = filepath.Base(os.Args[0])
+		WithService(filepath.Base(os.Args[0]))
 	}
 	if ver := os.Getenv("DD_VERSION"); ver != "" {
-		c.version = ver
+		WithServiceVersion(ver)
+	}
+	if analyticsStr := os.Getenv("DD_TRACE_ANALYTICS_ENABLED"); analyticsStr != "" {
+		if v, err := strconv.ParseBool(analyticsStr); err == nil {
+			WithAnalytics(v)
+		}
+	}
```

Comment:
In Go we have this idiom of
> The greater the distance between a name's declaration and its uses,
the longer the name should be.
> -- [_"What's in a name"_ by Andrew Gerrand](https://talks.golang.org/2014/names.slide#4)

In this case it's not needed to have such a long variable name as `analyticsStr`. I think we can totally drop that one and the error checking and just do:

```
if v, err := strconv.ParseBool(os.Getenv("DD_TRACE_ANALYTICS_ENABLED")); err == nil {
    WithAnalytics(v)(c)
}
```
It will be fine if the env. var. is empty.